daemon/tests/happy_path.rs

```diff
@@ -152,6 +154,46 @@ async fn collaboratively_close_an_open_cfd() {
     assert_next_state!(CfdState::Closed, maker, taker, order_id);
 }
 
+#[tokio::test]
+#[cfg_attr(not(feature = "fake_clock"), ignore)]
+async fn force_close_an_open_cfd() {
+    let _guard = init_tracing();
+
+    let start_time = OliviaData::example_0().announcement().expected_outcome_time
```

Comment:
🤔 I think the `fake_clock` feature flag is a well-executed and interesting solution, but I'm not totally sold.

The way I see it, we're introducing quite a bit of complexity to get past this check https://github.com/itchysats/itchysats/blob/90999363343d6321443aeb37a63aef04755a88da/daemon/src/oracle.rs#L159-L166 in this test. Couldn't we just make the timestamp in the event ID dynamic?

I'm reluctant to add something to production code that is there just to help us test things. If there is no valid alternative then we can merge this.

Reply:
This is not the only thing actually. With the fake clock, we are also working around how we compute what the next bitmex price event ID is when we send out a new offer for example.

Reply:
We're not adding anything except a forwarding call to our internal library (we could say we're adding one level of abstraction, but no functional changes whatsoever). Compared to the gymnastics we had to go through to allow testing maker and taker actors, it's IMHO neglegible (especially that we have clippy lint and it's a 1:1 correspondence with calls, so no mental overhead).

This is fairly lightweight solution, a different one that was contemplated was having a clock provider as an actor that can send messages with current time on a tick and also respond with current time to a message querying the time. The obvious downside would be all the cost of rolling out the actor's address into all the actors that need to ask for the time). The upside of that solution would be not needing a feature flag (we would only mock the clock provider in the test harness).

I'm afraid there's no easier approach here :(

Reply:
I definitely prefer what you've done over introducing a clock provider actor. If you guys reckon there is no simple way to handle this problem then I guess we should go with this.

Reply:
thanks to @luckysori trying out the changeset after @da-kami's refactor we don't need to rely on particular implementation of our lookup of the matching attestation, and we do not need to fake the clock anymore. check out the new simplified version :)